Source/JavaScriptCore/bytecode/PutByVariant.cpp

```diff
@@ -248,6 +267,35 @@ bool PutByVariant::attemptToMerge(const PutByVariant& other)
         return true;
     }
 
+    case CustomAccessorSetter: {
+        if (other.m_kind != CustomAccessorSetter)
+            return false;
+
+        if (m_customAccessorSetter != other.m_customAccessorSetter)
+            return false;
+
+        if (m_domAttribute || other.m_domAttribute) {
+            if (!(m_domAttribute && other.m_domAttribute))
+                return false;
+            if (*m_domAttribute != *other.m_domAttribute)
+                return false;
+        }
+
+        if (m_conditionSet.isEmpty() != other.m_conditionSet.isEmpty())
+            return false;
+
+        ObjectPropertyConditionSet mergedConditionSet;
```

Comment:
Shouldn't this be `mergedConditionSet = other.m_conditionSet`?

Reply:
Right, changed.

Reply:
Ah, no. We have `m_conditionSet.isEmpty() != other.m_conditionSet.isEmpty()`. So if m_conditionSet is empty, then other is also empty.